src/client/src/sync/mod.rs

```diff
@@ -135,6 +139,32 @@ mod context {
         }
     }
 
+    impl<S> StoreContext<S>
+    where
+        S: Spec + Send + Sync + 'static,
+        <S as Spec>::Status: Send + Sync,
+        S::IndexKey: Send + Sync,
+    {
+        pub fn watch<'a>(
+            &'a self,
+        ) -> impl Stream<Item = MetadataChanges<S, AlwaysNewContext>> + 'a {
+            let mut listener = self.store.change_listener();
+            let (sender, receiver) = async_channel::unbounded();
+
+            fluvio_future::task::spawn(async move {
+                loop {
+                    listener.listen().await;
+                    let changes = listener.sync_changes().await;
+                    if let Err(e) = sender.send(changes).await {
+                        error!("Failed to send Metadata update: {:?}", e);
+                    }
+                }
+            });
+
+            receiver
+        }
+    }
```

Comment:
@sehz this is the new implementation of the Watch Stream. It definitely yields items, but it does not seem to trigger every time a topic is created or deleted, but rather it triggers periodically. It seems like we should do _something_ to make sure this updates on every change properly.